typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,151 @@
+            this.absoluteFilepathHere = absoluteFilepath;
+            this.absoluteFilepathOnServiceContainer = absoluteFilepathOnServiceContainer;
+    }   
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getAbsoluteFilepathHere(): string { 
+        return this.absoluteFilepathHere;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getAbsoluteFilepathOnServiceContainer(): string { 
+        return this.absoluteFilepathOnServiceContainer;
+    }
+}
+
+
+
+class ServiceContext {
+    
+    private readonly client: ApiContainerServiceClient;
+    private readonly serviceId: ServiceID;
+    private readonly ipAddress: string;
+    private readonly enclaveDataVolMountpointHere: string;
+    private readonly enclaveDataVolMountpointOnServiceContainer: string;
+
+    constructor(
+        client: ApiContainerServiceClient,
+        serviceId: ServiceID,
+        ipAddress: string,
+        enclaveDataVolMountpointHere: string,
+        enclaveDataVolMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.enclaveDataVolMountpointHere = enclaveDataVolMountpointHere;
+            this.enclaveDataVolMountpointOnServiceContainer = enclaveDataVolMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, Uint8Array | string, Error] {
+        const serviceId: ServiceID = this.serviceId;
+        const args: ExecCommandArgs = newGetExecCommandArgs(serviceId, command);
+
+        var resp: ExecCommandResponse;
+        var error: grpc.ServiceError;
+        this.client.execCommand(args, function(error, feature){ //TODO (comment) - don't use return value, but use callback parameters
+            error = error; 
+            resp = feature;
+        });
+        
+        //TODO (comment) - Will this error checking propogate as needed (Kevin mentioned that returning error as-is is good enough)
+        if (error != null) {
+            return [0, null, error]; 
+            //TODO - passing in actual error, but no personalized message
+            //I was thinking of `new Error("An error occurred executing command " + command +  " on service " + serviceId + 
+            //". Here is the error message: " + error)` but don't think this propogates the error as needed, just instantiates an error
+        }
+        return [resp.getExitCode(), resp.getLogOutput(), null];
```

Comment:
Just be aware that, pursuant to above, this will probably throw a null pointer exception because (highly likely) that `resp.getExitCode()` will be called while `resp` is still null, before the `execCommand` actually comes back